bdk_core/src/chain_graph.rs

```diff
@@ -63,8 +63,37 @@ impl<I: ChainIndex> ChainGraph<I> {
         &self,
         update: &Self,
     ) -> Result<ChangeSet<I>, sparse_chain::UpdateFailure<I>> {
+        let mut chain_changeset = self.chain.determine_changeset(&update.chain)?;
+
+        // If we are invalidating checkpoints, it signals that a reorg has occured. We modify the
+        // changeset to add back "invalidated" transactions which do not conflict with transactions
+        // introduced by the `update`. We assume that all unconfirmed transactions in the `update`
+        // is included in the mempool.
+        if let Some(invalid_from) = chain_changeset.invalid_from {
```

Comment:
It seems that we can easily get `invalid_from` from the checkpoint changes rather than having an explicit field for this.